gunicorn/sock.py

```diff
@@ -20,6 +20,7 @@
 class BaseSocket(object):
 
     def __init__(self, address, conf, log, fd=None):
+        self.parent = os.getpid()
```

Comment:
right. removing it. About systemd, we should handle it in another commit.. The genertal idea here would be keeping in env the systemd fds that were given to us. So the `create_sockets` method would return a the listeners and external fds. I will submit a patch.